crash_tracker: print more context on failure

pgellert opened the pull request:
Add more context into crash recording messages:
* [crash_tracker: record context of crash tracker write](https://github.com/redpanda-data/redpanda/commit/bbd13462a6a76b22b95a579349f8633d0e63a422)
* [crash_tracker: record context of write skipping](https://github.com/redpanda-data/redpanda/commit/aee8be1d90b1254da42ccb04b4030f0655b688c2)

Example:
```
Recorded crash reason to crash file on shard 0 (SIGILL)
```

Also detect the crash recording message as a `NodeCrash`-type failure in ducktape.

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [ ] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [x] v25.2.x
- [x] v25.1.x
- [ ] v24.3.x

## Release Notes

* none

<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->

pgellert committed:
crash_tracker: record context of crash tracker write

pgellert committed:
crash_tracker: record context of write skipping

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/74517-->
<!--4be90043-ea61-485b-b117-5c0b6a940f1e-->
<!--65dff738-4563-4140-a5e3-c3589f7b61bb-->
<!--7bea166f-f014-4afc-9d92-d52215e980da-->
<!--7fc8237a-9df9-4103-afb0-212b5606fb33-->
<!--6ca8e878-144b-4550-a9fb-68536d4ca777-->
<!--093246b4-d43b-4319-9dfa-72e4875abc12-->
<!--f4399cd1-79b4-4888-bf24-4b3d78848ac4-->
<!--title-->
## Retry command for [Build#74517](https://buildkite.com/redpanda/redpanda/builds/74517)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/cluster_bootstrap_test.py::ClusterBootstrapNew.test_misconfigured_root_driven_bootstrap
tests/rptest/tests/crash_loop_checks_test.py::CrashLoopChecksTest.test_vassert_message@{"signal_shard":1}
tests/rptest/tests/crash_loop_checks_test.py::CrashLoopChecksTest.test_vassert_message@{"signal_shard":0}
tests/rptest/tests/storage_failure_injection_test.py::StorageFailureInjectionTest.test_storage_failure_injection
tests/rptest/tests/cluster_linking_e2e_test.py::ShadowLinkingReplicationTests.test_replication_with_failures
```
<!--/body-->

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--3c2bd0a8-c0df-441d-8a88-b120895f8fd2-->
<!--TEST RESULTS-->
<!--d82b5a03-214e-4c58-b0a7-a5441e8e05b8-->
<!--TEST RESULTS-->
<!--7265ed68-3e87-4a00-8d42-e78689c53e5f-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/74517">build#74517</a></summary>


| test_class                                     | test_method                                    | test_arguments          | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  | test_history                                                                                                                                                                      |
|:-----------------------------------------------|:-----------------------------------------------|:------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ClusterBootstrapNew                            | test_misconfigured_root_driven_bootstrap       | null                    | integration | https://buildkite.com/redpanda/redpanda/builds/74517#019a01df-3b58-4eea-a0db-9b10e5c4dfad | FAIL          | 0/1      |                                                                                                                                                                         | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ClusterBootstrapNew&test_method=test_misconfigured_root_driven_bootstrap        |
| ClusterBootstrapNew                            | test_misconfigured_root_driven_bootstrap       | null                    | integration | https://buildkite.com/redpanda/redpanda/builds/74517#019a01f8-115e-4460-b1a3-ab5bbf664673 | FAIL          | 0/1      |                                                                                                                                                                         | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ClusterBootstrapNew&test_method=test_misconfigured_root_driven_bootstrap        |
| ShadowLinkingReplicationTests                  | test_replication_with_failures                 | null                    | integration | https://buildkite.com/redpanda/redpanda/builds/74517#019a01df-3b5a-4ab6-85ba-4c42e1807978 | FLAKY         | 11/21    | upstream reliability is '65.22911051212938'. current run reliability is '52.38095238095239'. drift is 12.84816 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingReplicationTests&test_method=test_replication_with_failures        |
| ShadowLinkingReplicationTests                  | test_replication_with_failures                 | null                    | integration | https://buildkite.com/redpanda/redpanda/builds/74517#019a01f8-1162-4b69-86e0-e84367552079 | FLAKY         | 2/21     | upstream reliability is '65.1978417266187'. current run reliability is '9.523809523809524'. drift is 55.67403 and the allowed drift is set to 50. The test should FAIL  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingReplicationTests&test_method=test_replication_with_failures        |
| CrashLoopChecksTest                            | test_vassert_message                           | {"signal_shard": 0}     | integration | https://buildkite.com/redpanda/redpanda/builds/74517#019a01f8-115b-4252-bc61-907aaf2904a3 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=CrashLoopChecksTest&test_method=test_vassert_message                            |
| CrashLoopChecksTest                            | test_vassert_message                           | {"signal_shard": 1}     | integration | https://buildkite.com/redpanda/redpanda/builds/74517#019a01f8-115d-4b4f-9234-2b365d825c79 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=CrashLoopChecksTest&test_method=test_vassert_message                            |
| StorageFailureInjectionTest                    | test_storage_failure_injection                 | null                    | integration | https://buildkite.com/redpanda/redpanda/builds/74517#019a01df-3b59-463f-905f-058757da11d6 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=StorageFailureInjectionTest&test_method=test_storage_failure_injection          |
| StorageFailureInjectionTest                    | test_storage_failure_injection                 | null                    | integration | https://buildkite.com/redpanda/redpanda/builds/74517#019a01f8-1160-4fcc-babd-3536f507cba1 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=StorageFailureInjectionTest&test_method=test_storage_failure_injection          |
| TxAtomicProduceConsumeTest                     | test_basic_tx_consumer_transform_produce       | {"with_failures": true} | integration | https://buildkite.com/redpanda/redpanda/builds/74517#019a01df-3b59-463f-905f-058757da11d6 | FLAKY         | 19/20    | upstream reliability is '99.1769547325103'. current run reliability is '95.0'. drift is 4.17695 and the allowed drift is set to 50. The test should PASS                | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=TxAtomicProduceConsumeTest&test_method=test_basic_tx_consumer_transform_produce |
| src/v/cluster/tests/shard_placement_table_test | src/v/cluster/tests/shard_placement_table_test |                         | unit        | https://buildkite.com/redpanda/redpanda/builds/74517#019a01a9-9372-48c4-9f60-9c0406760e4c | FAIL          | 0/1      |                                                                                                                                                                         |                                                                                                                                                                                   |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/74943">build#74943</a></summary>


| test_class                  | test_method                                             | test_arguments                                                                                                 | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 | test_history                                                                                                                                                                                   |
|:----------------------------|:--------------------------------------------------------|:---------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ShadowLinkBasicTests        | test_topic_properties_replicated                        | {"source_cluster_spec": {"cluster_type": "kafka", "kafka_quorum": "COMBINED_KRAFT", "kafka_version": "3.8.0"}} | integration | https://buildkite.com/redpanda/redpanda/builds/74943#019a1755-41a1-488f-b9f1-97c05d6fdc40 | FLAKY         | 18/21    | upstream reliability is '95.12578616352201'. current run reliability is '85.71428571428571'. drift is 9.4115 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkBasicTests&test_method=test_topic_properties_replicated                            |
| NodesDecommissioningTest    | test_decommissioning_finishes_after_manual_cancellation | {"cloud_topic": false, "delete_topic": true}                                                                   | integration | https://buildkite.com/redpanda/redpanda/builds/74943#019a1760-224f-476c-bac9-6c0d7841629f | FLAKY         | 20/21    | upstream reliability is '98.1941309255079'. current run reliability is '95.23809523809523'. drift is 2.95604 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=NodesDecommissioningTest&test_method=test_decommissioning_finishes_after_manual_cancellation |
| NodesDecommissioningTest    | test_decommissioning_finishes_after_manual_cancellation | {"cloud_topic": true, "delete_topic": true}                                                                    | integration | https://buildkite.com/redpanda/redpanda/builds/74943#019a1760-2251-47cf-bb3d-a57198f6f670 | FLAKY         | 20/21    | upstream reliability is '99.28400954653938'. current run reliability is '95.23809523809523'. drift is 4.04591 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=NodesDecommissioningTest&test_method=test_decommissioning_finishes_after_manual_cancellation |
| StorageFailureInjectionTest | test_storage_failure_injection                          | null                                                                                                           | integration | https://buildkite.com/redpanda/redpanda/builds/74943#019a1755-41a8-4d4d-a943-5a04c34d864e | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                 | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=StorageFailureInjectionTest&test_method=test_storage_failure_injection                       |
| StorageFailureInjectionTest | test_storage_failure_injection                          | null                                                                                                           | integration | https://buildkite.com/redpanda/redpanda/builds/74943#019a1760-2255-4200-b031-afaf080a2ce0 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                 | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=StorageFailureInjectionTest&test_method=test_storage_failure_injection                       |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75024">build#75024</a></summary>


| test_class                 | test_method                                             | test_arguments                                | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  | test_history                                                                                                                                                                                   |
|:---------------------------|:--------------------------------------------------------|:----------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| EndToEndCloudTopicsTxTest  | test_write                                              | null                                          | integration | https://buildkite.com/redpanda/redpanda/builds/75024#019a264d-1360-4604-9acd-3f0c608f5e76 | FLAKY         | 15/21    | upstream reliability is '96.21928166351607'. current run reliability is '71.42857142857143'. drift is 24.79071 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=EndToEndCloudTopicsTxTest&test_method=test_write                                             |
| MountUnmountIcebergTest    | test_simple_remount                                     | {"cloud_storage_type": 1}                     | integration | https://buildkite.com/redpanda/redpanda/builds/75024#019a260c-e44e-4523-bb22-359b63cc5394 | FLAKY         | 15/21    | upstream reliability is '85.51401869158879'. current run reliability is '71.42857142857143'. drift is 14.08545 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=MountUnmountIcebergTest&test_method=test_simple_remount                                      |
| NodesDecommissioningTest   | test_decommissioning_finishes_after_manual_cancellation | {"cloud_topic": false, "delete_topic": false} | integration | https://buildkite.com/redpanda/redpanda/builds/75024#019a260c-e448-4e4b-ad96-36fafa8e50cd | FLAKY         | 20/21    | upstream reliability is '98.74686716791979'. current run reliability is '95.23809523809523'. drift is 3.50877 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=NodesDecommissioningTest&test_method=test_decommissioning_finishes_after_manual_cancellation |
| TxAtomicProduceConsumeTest | test_basic_tx_consumer_transform_produce                | {"with_failures": true}                       | integration | https://buildkite.com/redpanda/redpanda/builds/75024#019a260c-e44e-4523-bb22-359b63cc5394 | FLAKY         | 19/21    | upstream reliability is '100.0'. current run reliability is '90.47619047619048'. drift is 9.52381 and the allowed drift is set to 50. The test should PASS              | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=TxAtomicProduceConsumeTest&test_method=test_basic_tx_consumer_transform_produce              |
</details>
<!--/body-->

pgellert committed:
dt: treat crash recording success as crash

This is especially valuable for release-mode CI test failures that crash
on SIGILL, because then the only log line that currently gets printed is
the "Recorded crash reason..." log line.

pgellert commented:
[force-push](https://github.com/redpanda-data/redpanda/compare/9b68d6a5b7191dc6f30c6516fe37942efd557644..e8293a47dea96ee96210960ecca8f218501394c4): adjust failing ducktape tests based on the new crash log detection + address review comment

pgellert commented on tests/rptest/services/redpanda.py:
```diff
@@ -4113,7 +4113,7 @@ def is_allowed_log_line(line: str) -> bool:
             crash_log = None
             for line in node.account.ssh_capture(
-                f"grep -e SEGV -e Segmentation\\ fault -e [Aa]ssert -e Sanitizer -e 'Aborting on shard' {RedpandaService.STDOUT_STDERR_CAPTURE} || true",
+                f"grep -e SEGV -e Segmentation\\ fault -e [Aa]ssert -e Sanitizer -e 'Aborting on shard' -e 'Recorded crash reason' {RedpandaService.STDOUT_STDERR_CAPTURE} || true",
```
Yeah, makes sense

pgellert commented on tests/rptest/tests/cluster_bootstrap_test.py:
```diff
@@ -61,8 +62,12 @@ def test_misconfigured_root_driven_bootstrap(self):
             assert False, "Should have been unable to start"
-        except ducktape.errors.TimeoutError:
-            # The cluster should be unable to start.
+        except NodeCrash as e:
+            # The cluster should be unable to start, and node 0 should shut down during startup
+            assert len(e.crashes) == 1, f"Unexpected crashes: {e.crashes}"
+            assert e.crashes[0][0] == self.redpanda.nodes[0], (
+                f"Unexpected crashes: {e.crashes}"
+            )
```
Here, what happens is that node 0 "crashes" with an exception during startup, while the rest of the nodes are just stuck during startup. I've considered changing to `expect_fail`, but that would expect all of the nodes to crash, but here only node 0 crashes. So I've opted to just change the expected exception in this case to match the behaviour of `self.redpanda.start` in this test scenario.

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/74943-->
<!--9ef834f9-29c8-4aac-9214-679ed811bff9-->
<!--451250d3-797e-460d-9d59-61bcfaf53271-->
<!--title-->
## Retry command for [Build#74943](https://buildkite.com/redpanda/redpanda/builds/74943)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/storage_failure_injection_test.py::StorageFailureInjectionTest.test_storage_failure_injection
```
<!--/body-->

pgellert commented on tests/rptest/services/redpanda.py:
```diff
@@ -279,6 +279,7 @@ class CloudStorageCleanupStrategy(enum.Enum):
     ),
     re.compile("assert - Backtrace:"),
     re.compile("finject - .* flush called concurrently with other operations"),
+    re.compile("crash reason to crash file.*(vassert)"),
```
This is needed because `tests/rptest/tests/storage_failure_injection_test.py::StorageFailureInjectionTest.test_storage_failure_injection` can have vassert's after failures injected at the storage layer.

pgellert commented:
[force-push](https://github.com/redpanda-data/redpanda/compare/e8293a47dea96ee96210960ecca8f218501394c4..aec257e5cf2c4aad31d50d8749aa8e26dd80d3c4): also fix the test failure for `StorageFailureInjectionTest.test_storage_failure_injection`

vbotbuildovich commented:
/backport v25.2.x

vbotbuildovich commented:
/backport v25.1.x
